docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,406 @@
+
+### Receiving community history archives
+
+Community member nodes go through the following (high level) process to fetch and restore community message histories:
+
+1. User joins community and becomes community member (see [org channels spec](https://github.com/status-im/specs/pull/151))
+2. By joining a community, member nodes automatically subscribe to special channel for message archive metadata exchange provided by the community
+3. Member node requests live message history (last 30 days) of special community channel from store nodes (in addition to last 7 days of message for all other community channels)
+4. Member node receives waku message ([14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/)) that contains the metadata magnet link from the special channel
+5. Member node extracts the magnet link from the waku message and passes it to torrent client
+6. Member node downloads [message archive index](#message-history-archive-index) file and determines which message archives are not downloaded yet (all or some)
+7. Member node fetches missing message archive data via torrent
+8. Member node unpacks and decompresses message archive data to then hydrate its local database, deleting any messages for that community that the database previously stored in the same time range as covered by the message history archive
+
+## Storing live messages
+
+Community owner nodes MUST store live messages as [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/). This is required to provide confidentiality, authenticity, and integrity of message data distributed via the BitTorrent layer, and later validated by Status nodes when they unpack message history archives.
+
+Community owner nodes SHOULD remove those messages from their local databases once they are older than 30 days and after they have been turned into message archives and distributed to the BitTorrent network.
+
+## Exporting messages for bundling
+
+Community owner nodes export messages from their local database for creating and bundling history archives using the following criteria:
+
+- Messages to be exported MUST have a `contentTopic` that match any of the topics of the community channels
+- Messages to be exported MUST have a `timestamp` that lies within a time range of 7 days
+
+The `timestamp` is determined by the context in which the community owner node attempts to create a message history archives as described below:
+
+1. The community owner node attempts to create an archive periodically for the past seven days (including the current day). In this case, the `timestamp` has to lie within those 7 days.
+2. The community owner node has been offline (owner node's main process has stopped and needs restart) and attempts to create archives for all the live messages it has missed since it went offline. In this case, the `timestamp` has to lie within the day the latest message was received and the current day.
```

Comment:
Shouldn't the whole missed time range be broken into 7-day intervals? I think item 1 already covers it and there is no need for the second item. 
I'd suggest adding the explanation of item 2 (how to measure the missed time range) under the second time of this section https://github.com/status-im/specs/blob/c9184b74b5623fe14aa4bd99ed8b268a65984117/docs/raw/serving-community-history.md#serving-archives-for-missed-messages

Reply:
Okay, will update this. I thought it would be good to be explicit about the scenario that the node can be offline, in which case there can be up to 30 days worth of messages from which the node needs to create archives (4, 1 for each 7 days)